From 0ad148040da972a977d2cb7f8b0a1af7a3df038d Mon Sep 17 00:00:00 2001
From: hanzongcheng <hanzongcheng@huawei.com>
Date: Mon, 10 Oct 2022 05:05:42 +0800
Subject: [PATCH] ethercat: Fix ethercat tool compilation

* After commit f3ccfda193197 ("ethtool: extend coalesce setting uAPI with CQE mode") and
  commit 7462494408cd3 ("ethtool: extend ringparam setting/getting API with rx_buf_len"),
  ethtool extend the setting/getting API, it causes the following compilation error:

    ethtool-5.10-ethercat.c: error: initialization of '...' from incompatible pointer type '...' [-Werror=incompatible-pointer-types]
    .get_ringparam  = e1000_get_ringparam,
                      ^~~~~~~~~~~~~~~~~~~
    .set_ringparam  = e1000_set_ringparam,
                      ^~~~~~~~~~~~~~~~~~~
    .get_coalesce  = e1000_get_coalesce,
                     ^~~~~~~~~~~~~~~~~~
    .set_coalesce  = e1000_set_coalesce,
                     ^~~~~~~~~~~~~~~~~~
  Modify these function arguments to fix this error.

Signed-off-by: hanzongcheng <hanzongcheng@huawei.com>

diff --git a/devices/e1000e/ethtool-5.10-ethercat.c b/devices/e1000e/ethtool-5.10-ethercat.c
index 3e71855e..c03eff67 100644
--- a/devices/e1000e/ethtool-5.10-ethercat.c
+++ b/devices/e1000e/ethtool-5.10-ethercat.c
@@ -655,7 +655,9 @@ static void e1000_get_drvinfo(struct net_device *netdev,
 }
 
 static void e1000_get_ringparam(struct net_device *netdev,
-				struct ethtool_ringparam *ring)
+				struct ethtool_ringparam *ring,
+				struct kernel_ethtool_ringparam *kernel_ring,
+				struct netlink_ext_ack *extack)
 {
 	struct e1000_adapter *adapter = netdev_priv(netdev);
 
@@ -666,7 +668,9 @@ static void e1000_get_ringparam(struct net_device *netdev,
 }
 
 static int e1000_set_ringparam(struct net_device *netdev,
-			       struct ethtool_ringparam *ring)
+			       struct ethtool_ringparam *ring,
+			       struct kernel_ethtool_ringparam *kernel_ring,
+			       struct netlink_ext_ack *extack)
 {
 	struct e1000_adapter *adapter = netdev_priv(netdev);
 	struct e1000_ring *temp_tx = NULL, *temp_rx = NULL;
@@ -1991,7 +1995,9 @@ static int e1000_set_phys_id(struct net_device *netdev,
 }
 
 static int e1000_get_coalesce(struct net_device *netdev,
-			      struct ethtool_coalesce *ec)
+			      struct ethtool_coalesce *ec,
+			      struct kernel_ethtool_coalesce *kernel_coal,
+			      struct netlink_ext_ack *extack)
 {
 	struct e1000_adapter *adapter = netdev_priv(netdev);
 
@@ -2004,7 +2010,9 @@ static int e1000_get_coalesce(struct net_device *netdev,
 }
 
 static int e1000_set_coalesce(struct net_device *netdev,
-			      struct ethtool_coalesce *ec)
+			      struct ethtool_coalesce *ec,
+			      struct kernel_ethtool_coalesce *kernel_coal,
+			      struct netlink_ext_ack *extack)
 {
 	struct e1000_adapter *adapter = netdev_priv(netdev);
 
-- 
2.34.1

